__tests__/app.test.js

```diff
@@ -189,6 +201,24 @@ describe("/api/articles", () => {
           );
         });
     });
+
+    test("GET 404: responds with 404 for non-existent topic", () => {
+      return request(app)
+        .get("/api/articles?topic=non_existent_topic")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("404 - Not Found: topic  not found");
+        });
+    });
+
+    test("GET 404: responds with 400 if topic value missing", () => {
+      return request(app)
+        .get("/api/articles?topic=")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("400 - Bad Request: Topic value missing");
+        });
+    });
```

Comment:
Let's build a test for when the topic is in the db but has no articles like for the topic paper. We should receive a 200 and an empty array :)

Reply:
Great idea! Thank you again. :)